doc/nrf/app_dev/device_guides/nrf54h/ug_nrf54h20_suit_recovery.rst

```diff
@@ -236,3 +236,85 @@ To turn an application into a recovery application, the following steps have to
+
+.. note::
+   To exit the recovery mode, a device firmware update must be performed.
+
+There are two variants of this feature:
+
+* Recovery button checked in the recovery application run as a companion image at boot stage (recommended).
+* Recovery button checked in the main application.
+
+The first variant is recommended as it allows to perform the check independently of the main application.
+
+Recovery button checked in the recovery application
+===================================================
+
+In this variant the recovery application is run as a companion image before the main application when the device boots.
+It checks if the device already is in recovery mode.
+If it is, the recovery application proceeds with its normal operation allowing to recover firmware.
+
+If the device is not in recovery mode, the recovery application checks if the specified button is pressed.
+If the button is pressed, the recovery application sets the recovery flag and reboots the device.
+Otherwise it sends a confirmation message to the Secure Domain Firmware, which then halts the recovery application and proceeds with booting the main application.
+
+To enable the feature in this variant perform the following steps:
+
+1. Add the following code to :file:`sysbuild/recovery.overlay` in the main application directory:
+
+   .. code-block:: dts
+
+      / {
+         chosen {
+            ncs,recovery-button = &button0;
+         };
+      };
+
+   Replace ``button0`` with the appropriate button node.
+
+2. Enable :kconfig:option:`CONFIG_SUIT_RECOVERY_BUTTON` in the *recovery application* configuration.
+
+   When building from the command line, this can be done by adding ``-Drecovery_CONFIG_SUIT_RECOVERY_BUTTON=y`` to the build command.
+
+.. note::
+   In this option running the recovery application as a companion image is orchestrated by the root SUIT manifest.
+   Special care must be taken if the root manifest template is modified.
+   If incorrect modifications are made to the root manifest, the button press feature might not work.
+   This might lead to the device being unrecoverable without a JLink connection in case of a crash.
+
+
+Recovery button checked in the main application
+===============================================
+
+In this variant a check if the specified button is pressed is performed at an early stage when inside the main application firmware.
+If the button is pressed, the recovery flag is set and the device reboots.
+
+To enable the feature in this variant perform the following steps:
+
+1. Add the following overlay to the main application's configuration:
+
+   .. code-block:: dts
+
+      / {
+         chosen {
+            ncs,recovery-button = &button0;
+         };
+      };
+
+   Replace ``button0`` with the appropriate button node.
+
+2. Enable :kconfig:option:`CONFIG_SUIT_RECOVERY_BUTTON` in the main application configuration.
```

Comment:
Just thinking - since `SUIT_RECOVERY_BUTTON` already depends on DTS: `$(dt_chosen_enabled,$(DT_CHOSEN_NCS_RECOVERY_BUTTON))`, maybe we can state `default y` in that symbol too, so the DTS modification will be the only one required to enable this feature?

In other words - is there a practical case, in which the `ncs,recovery-button` is defined, yet the recovery button should not be checked?

Reply:
You are probably right, I've changed this in the other PR and removed the second step